src/appengine/libs/issue_management/jira/issue_tracker_manager.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Issue tracker manager functions."""
+from __future__ import absolute_import
+
+from builtins import object
+import jira
+import json
+
+from config import db_config
+
+
+class IssueTrackerManager(object):
+  """Issue tracker manager."""
+
+  def __init__(self, project_name):
+    """"Construct an issue tracker manager instance based on parameters."""
+    self._client = None
+    self.project_name = project_name
+
+  @property
+  def client(self):
+    """HTTP Client."""
+    if self._client is None:
+      self._client = self._create_client()
+
+    return self._client
+
+  def _create_client(self):
+    """Return a client object for querying the issue tracker."""
+    config = db_config.get()
+    credentials = json.loads(config.jira_credentials)
+    jira_url = config.jira_url
+    jira_client = jira.JIRA(
+        jira_url, auth=(credentials['username'], credentials['password']))
+    return jira_client
+
+  def save(self, issue):
+    """Save an issue."""
+    self._update(issue)
+
+  def create(self):
+    """Create an issue."""
+    jira_project = self.project_name
+    default_fields = {
+        'project': jira_project,
+        'summary': 'Default summary',
+        'description': 'Default description',
+        'issuetype': {
+            'name': 'Bug'
+        }
+    }
+
+    jira_issue = self.client.create_issue(fields=default_fields)
+    return jira_issue
+
+  def _update(self, issue):
+    """Update an issue."""
+
+    # Get labels from LabelStore.
+    labels = []
+    for label in issue.labels:
+      labels.append(label)
+
+    # Get components from LabelStore.
+    components = []
+    for component in issue.components:
+      components.append(component)
+
+    # Get watchers from LabelStore.
+    watchers = []
+    for watcher in issue.ccs:
+      watchers.append(watcher)
+
+    update_fields = {
+        'summary': issue.title,
+        'description': issue.body,
+        'labels': labels,
+        'components': components,
+    }
+
+    # Only add status if it has changed.
+    # Brittle - we should be pulling the equivalent of 'new' from the policy.
+    if issue.status != 'NOT STARTED':
```

Comment:
are these statuses the same for all jira projects? Or can they be different?

Reply:
There is a default set of statuses that ships with Jira, but you can also define new ones or edit the defaults as an administrator. Different projects may have different statuses available to them.

Reply:
I guess the other open item to add to my list above is to figure out how to grab the list of possible statuses for a given job.

Reply:
I'm assuming these are the default then? Alternatively, could we just always update the status regardless of the existing value, just to avoid this brittleness?

Reply:
Oops, rolled my comment meant for this into https://github.com/google/clusterfuzz/pull/1321#discussion_r369697191